src/librustc_resolve/lib.rs

```diff
@@ -2605,6 +2605,106 @@ impl<'a> Resolver<'a> {
                     mk_struct_span_error(false)
                 };
 
+                err.span_label(ident.span, &format!("this {} is private", binding.res().descr()));
+
+                // Display the chain of re-exports through to the original def for cases where the
```

Comment:
This below is adding some nearly 100 LOC to this method. It would be better if extracted to its own method.

Reply:
done

Reply:
Done